git-cliff-core/src/repo.rs

```diff
@@ -510,6 +518,73 @@ impl Repository {
     }
 }
 
+/// Compares two version strings semantically.
+///
+/// This function attempts to parse version strings and compare them semantically
+/// rather than alphabetically. It handles common version formats like:
+/// - v1.2.3
+/// - 1.2.3
+/// - v1.2.3-alpha.1
+///
+/// If semantic parsing fails, it falls back to alphabetical comparison.
+fn semantic_version_compare(a: &str, b: &str) -> std::cmp::Ordering {
+    use std::cmp::Ordering;
+
+    // Helper function to extract version parts and pre-release info
+    let parse_version = |version: &str| -> Option<(Vec<u64>, Option<String>)> {
```

Comment:
Hmm, `Option<(Vec<u64>, Option<String>)>` is a bit complex. Do you think defining an inner struct would make the code better in terms of readability? 🤔